src/algebra/finite_map/map_macro.rs

```diff
@@ -8,6 +8,7 @@ macro_rules! finite_map {
 
     // Create a map from elements
     ($($key:expr => $value:expr),+ $(,)?) => {{
+        use $crate::algebra::MapBuilder;
```

Comment:
Github won't let me comment on it, but I'd instead change this macro arm to have
```rust
$( <$crate::algebra::FiniteHashMap<_, _> as $crate::algebra::MapBuilder>::increment(&mut map, &$key, $value); )+
```
instead of
```rust
$( $crate::algebra::FiniteHashMap::increment(&mut map, &$key, $value); )+
```

Reply:
yes, I tried that, and it works, but for some reason this impacts type inference and the compiler asks for type annotations wherever I use the macro

Reply:
I take it back, it works now.  Maybe something else was broken last time I tried this.

Reply:
Hum, that's very strange